packages/grafana-ui/src/components/uPlot/plugins/KeyboardPlugin.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useLayoutEffect } from 'react';
+import uPlot from 'uplot';
+import { UPlotConfigBuilder } from '../config/UPlotConfigBuilder';
+
+interface KeyboardPluginProps {
+  config: UPlotConfigBuilder; // onkeypress, onkeyup, onkeydown (triggered by vizlayout handlers)
+}
+
+const PIXELS_PER_MS = 0.1 as const;
+const SHIFT_MULTIPLIER = 2 as const;
+const KNOWN_KEYS = new Set(['ArrowRight', 'ArrowLeft', 'ArrowUp', 'ArrowDown', 'Shift', ' ']);
+
+function minMax(n: number, min: number, max: number) {
+  return Math.min(Math.max(min, n), max);
+}
```

Comment:
**Nit:** I would probably just use the actual `Math.min(Math.max(min, n), max);` logic inline

The reason being is that I think the actual logic is not that much longer or harder to read then the extracted function call, on the other hand it communicates it more clearly what it is doing (`minMax()` with three arguments can take more time to grasp, at least to me).

Reply:
i'd just rename the function to `clamp(n, min, max)`. imo keeping it detached & named reads much easier than inline.

Reply:
Yea, `clamp` is actually a good name!

(we could also just [`_.clamp`](https://lodash.com/docs/4.17.15#clamp) in that case, as we already have `Lodash` as a dependency.)